packages/next/src/server/render.tsx

```diff
@@ -266,6 +266,7 @@ export type RenderOptsPartial = {
   domainLocales?: readonly DomainLocale[]
   disableOptimizedLoading?: boolean
   supportsDynamicResponse: boolean
+  botType?: 'headless' | 'html-limited' | undefined
```

Comment:
maybe a nit but it might help to classify the bots as `DOM` and `HTML`. In this characterization DOM bots take the response and produce a DOM (including running javascript etc...) and HTML bots simply consume the HTML itself without interpreting it.

I think that's kind of what headless means but its not really important that the bot is headless (no UI) just that it actually parses the HTML into DOM and then interprets the DOM not the HTML itself.